### PR TITLE
Add TektonConfig CRD to install Tekton Component

### PR DESCRIPTION
**Pull request by savitaashture**
# Changes

* Added new CRD called `TektonConfig`
   `TektonConfig` CRD provide options to user to choose which tekton component they want to install based on `profile`
* Updated doc for more info on `profile`
* Added API and Reconciler
* Added e2e and unit test

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
  Added new CRD called `TektonConfig` to install TektonCompnents
```

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -44,6 +44,26 @@ If the files in `pkg/apis` are updated we need to run `codegen` scripts
...
+Operator provides an option to choose which components needs to be installed by specifying `profile`.
+
+`profile` is an optional field and supported `profile` are
+* **basic** 
+* **default**
+* **all**
+
+1. If profile is `basic` **TektonPipeline** will be installed
+1. If profile is `default` or `" "` **TektonPipeline** and **TektonTrigger** will be installed
+1. If profile is `all` then all the Tekton Components installed
```
Comment (vdemeester) [inline docs/README.md:57]: Will the profiles be a bit different for the kubernetes platform vs openshift platform ? 
aka What would be the profile for what is done by OpenShift Pipelines as of today ?

Comment (nikhil-thomas) [reply docs/README.md:57]: extension magic. :tada: 
for example
`all` profile installs addons on openshift (no dashboard), but on k8s it installs dashboard but no addons.

the parts which are not common are handled as extensions.
https://github.com/tektoncd/operator/pull/179/files#diff-a100309e66fd0aa1063e2e110fd0d6d97eaa6516bf10a9f61285828a7a0697c1
and
https://github.com/tektoncd/operator/pull/179/files#diff-9180de7e95281f01fe01e488cdcc0c7119c98362aa6a33308095057d2cc6918c

Comment (savitaashture) [reply docs/README.md:57]: profile is common for both the openshift and kubernetes 

If `profile` is **all** 
On Openshift -> Pipeline, Triggers and Addons will be installed
On Kubernetes -> Pipeline, Triggers and Dashboard will be installed

**Comment by savitaashture** [comment]
> 2\. is is absolutely necessary to pass the client to the PostReconcile method in `Extension` interface `PostReconcile(context.Context, v1alpha1.TektonComponent, operatorv1alpha1.OperatorV1alpha1Interface) error` . can we initialize the client(s) inside the implementaitons as we are passing context. It is fine if there is no way around it.

@nikhil-thomas tried with this function [Get(ctx)](https://github.com/tektoncd/operator/blob/main/pkg/client/injection/client/client.go#L42) function  but its always giving [error](https://github.com/tektoncd/operator/blob/main/pkg/client/injection/client/client.go#L46) somehow context don't have the info when i call from other .go files but same thing work inside [controller.go](https://github.com/tektoncd/operator/blob/main/pkg/reconciler/kubernetes/tektonpipeline/controller.go)

**Comment by savitaashture** [comment]
Thank you :+1: 

PR is ready for another review

**Review by vdemeester — APPROVED** [review]

**Pull request merged**